### DISCUSSION TITLE
generating celltype specific TF to gene table

### DISCUSSION THREAD

**Discussion by sid5427** [aertslab/scenicplus#126] [Q&A] [answered]
Hi Seppe and community,

I am bit confused on how to generate a simple TF to GENE (or TF to Region to Gene)  connection table, for each celltype for my dataset. We want to intersect with results from other TF-gene networks generated from the same dataset.
The only place I see celltype specific mapping is in the heatmap-dotplot section and in the network plotting section where the graph is laid out from the nx_tables - creating the edges from TF to region and region to gene relationships.
 
I am hoping to get a table for **each celltype** which has columns something like this - TFname, Regioncoord, Gene_name, TF2R_score, T2G_score (TFname, Regioncoord and Gene_name values can be duplicates if we have multiple connections)

Could you please give an example of how to do this? maybe a celltype (say - "B_cells_1, B_cells_2, etc/") from the 10x multiome tutorial?

-- Sid

**Comment by SeppeDeWinter** [comment, marked as answer]
Hi @sid5427 

This data is stored in the following field of the `scplus_obj`: `scplus_obj.uns['eRegulon_metadata']`. However, this is not subdivided per cell type as the inference is run on the entire dataset (i.e. in a cell type agnostic manner). Getting information per cell type is not trivial. One way you could do this is by looking at gene expression and region accessibility of predicted target genes/regions in each cell type for each regulon. This is also what the dotplot aims to summarise. 
Unfortunately there is not automated way of stratifying regulons based on cell type. 

Best, 

Seppe

**Reply by sid5427** [reply]
Hi Seppe,

Thanks for your reply!

Ah -  I was worried about that. I was able to get a table from the heatmap_dotplot.data slot and it does look something similar to what I need.

I have question on how these two matrices are scaled from 0 to 1
```
size_matrix =scplus_obj.uns['eRegulon_AUC_filtered']['Region_based'] #specify what to plot as dot sizes, target region enrichment in this case
color_matrix = scplus_obj.to_df('EXP'), #specify  what to plot as colors, TF expression in this case
```

Could you clarify how the scaling works? is it some sort of normalization or log odds ratio? 

- Sid

**Reply by SeppeDeWinter** [reply]
It's a very simple scaling between 0 and 1, like this:

(x - min(x)) / (max(x) - min(x))

Best,

Seppe

**Comment by paupaiz** [comment]
@SeppeDeWinter could you please explain why the eRegulon specificity score (eRSS)
(defined as the specificty of eRegulons in clusters of cells) is not a way of stratifying regulons based on cell type? what @sid5427 was looking for.
<img width="977" alt="Screenshot 2024-08-04 at 7 27 08 AM" src="https://github.com/user-attachments/assets/461dd6c4-5ffb-4a7f-bbc4-5d654abce3f7">

**Reply by uqnsarke** [reply]
following

**Reply by SeppeDeWinter** [reply]
hi @paupaiz and @uqnsarke 

Indeed, you can use the RSS to select eRegulons per cell type.
However, the target genes and regions of the regulons are not cell type specific (that's what Sid was asking for).

For example.

Let's we have cell type A where TF1 together with TF2 regulates the expression of genes {a, b, c}
and we have cell type B where TF1 together with TF3 regulates the expression of genes {d, e, f}.

SCENIC+ would predict that {a, b, c, d, e, f} are target genes of TF1 and we don't have a way to select only the target genes of TF1 in cell type A for instance.

Does this make sense?

All the best,

Seppe

**Comment by zhli12** [comment]
Thanks for developing and maintaining this excellent package! I also have a related question. Would it make sense to subset the data by cell-type before running the scenicplus pipeline to obtain cell-type specific results, or would it cause negative impact due to reduced heterogeneity in the training data?  I'm wondering whether you might have any suggestions on the practice @SeppeDeWinter
